gusto/state.py

```diff
@@ -356,12 +359,13 @@ def dump(self, t=0, pickup=False):
                 self.pointdata_output.dump(self.fields, t)
 
             # Dump all the fields to the checkpointing file (backup version)
-            for field in self.to_pickup:
-                self.chkpt.store(field)
-            self.chkpt.write_attribute("/", "time", t)
-            # Create second backup file
-            copyfile(path.join(self.dumpdir, "chkpt.h5"),
-                     path.join(self.dumpdir, "chkptbk.h5"))
+            if self.output.checkpoint:
+                for field in self.to_pickup:
+                    self.chkpt.store(field)
+                self.chkpt.write_attribute("/", "time", t)
+                # Create second backup file
+                copyfile(path.join(self.dumpdir, "chkpt.h5"),
+                         path.join(self.dumpdir, "chkptbk.h5"))
```

Comment:
This copy should be done on a single rank.  Also, I'm not sure you're guaranteed filesystem consistency when you call this copy.

Reply:
I'm not sure I understand filesystem consistency. Do you mean this as in "what happens if the system crashes while the copying is taking place, could both files then be lost"? If this is the case then I don't really know how to rewrite the code other than how it was originally, with two files being written to seperatly.